modules/learn/pages/buckets-memory-and-storage/storage-engines.adoc

```diff
@@ -0,0 +1,66 @@
+= Storage Engines
+:description: pass:q[Couchbase supports two different backend storage mechanisms: Couchstore and Magma. It is important to understand which backend storage is best suited to your requirements.]
+
+[abstract]
+{description}
+
+== Couchstore
+
+Couchstore is the default bucket storage engine that has been in use for more than ten years. 
+It's optimised for high performance with large data sets while using fewer system resources (the minimum bucket size for Couchstore backend is 100 MB). 
+If you have a small data set that can fit in memory, then you should consider using Couchstore.
+
+[#storage-engine-magma]
+== Magma
+
+Magma is Couchbase's latest storage engine that is designed for high performance with very large datasets that do not fit in memory. It is ideal for use cases that rely primarily on disk access. The performance of disk access will be as good as the underlying disk subsystems. e.g. using NVMe SSDs will give higher performance.  
+Magma can work with very low amounts of memory for large datasets. e.g. for a node holding 5TB of data, Magma can be used with only 64GB RAM. Magma buckets do require a minimum of 256MB RAM and you should not go below a RAM allocation of 1% of your total data size. 
+It is especially suited to data sets that will not fit into available memory.
+
+IMPORTANT: Currently, Magma does not support Full Text Searching, Eventing or Analytics. If the use case requires these features, Couchstore should be used.
+
+== Couchstore and Magma at a Glance
```

Comment:
We should add one more entry to this table.
Minimum memory to data ratio. 
- Couchstore 10%
- Magma 1%